community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreFile.java

```diff
@@ -298,16 +297,26 @@ private static void setStoreVersionTrailer( FileSystemAbstraction fs, File targe
                 throw new IllegalStateException( "Required file missing: " + targetStoreFileName );
             }
         }
-        try ( StoreChannel fileChannel = fs.open( targetStoreFileName, "rw" ) )
+        try ( PagedFile pagedFile = pageCache.map( targetStoreFileName, pageCache.pageSize() ) )
+        {
+            long trailerOffset =
+                    getTrailerOffset( pagedFile, versionTrailer.split( " " )[0] );
+            if ( trailerOffset != -1 )
+            {
+                writeTrailer( pagedFile, trailer, trailerOffset );
+            }
+        }
+    }
+
+    private static boolean storeExists( PageCache pageCache, File targetStoreFileName )
+    {
+        try ( PagedFile file = pageCache.map( targetStoreFileName, pageCache.pageSize() ) )
         {
-            fileSize = fileChannel.size();
-            fileChannel.position( fileChannel.size() - trailer.length );
-            fileChannel.write( ByteBuffer.wrap( trailer ) );
         }
-        catch ( IllegalArgumentException e )
+        catch ( IOException e )
         {
-            throw withMessage( e, e.getMessage() + " | " + "size:" + fileSize + ", trailer:" + trailer.length +
-                    " for " + targetStoreFileName );
+            return false;
         }
+        return true;
```

Comment:
Not that I know of. What we want to find out is if it is mappable by the page cache, and the way to do that is to map it.